Bubble background color and chat message row background color

samuelo-101 opened the pull request:
#16 
- Added 2 additional attributes 'backgroundRcv' and 'backgroundSend' that will allow devs to specify the background color of the entire row of a received and/or sent message
- Passing the 2 new variables from ChatView, through to the MessageViewHolder
- Altered and added code in method setBackground of MessageViewHolder to apply a background color to both the bubble and entire row of sent/received messages
- Added the 2 new attributes to activity_main.xml to demonstrate their use
- Added a 'SENT' message with 1.5 seconds delay to MainActivity

samuelo-101 committed:
Issue 16:
- Added 2 additional attributes 'backgroundRcv' and 'backgroundSend' that will allow devs to specify the background color of the entire row of a received and/or sent message
- Passing the 2 new variables crom ChatView, through to the MessageViewHolder
- Altered and added code in method setBackground of MessageViewHolder to apply background color to both the bubble and entire row of sent/received messages
- Added the 2 new attributes to activity_main.xml to demonstrate their use
- Added a 'SENT' message with 1.5 seconds delay to MainActivity

MichaelObi reviewed (changes requested):
Hello Samuel,
Thanks a lot for helping out.
There's a bit of a problem building your PR because we can't find the `backgroundRcv` and `backgroundSend` Identifiers. They were taken out of `attrs.xml`. I'll get on with the review as soon as I can build it.

samuelo-101 committed:
- Removed 'backgroundRcv' and 'backgroundSend' attributes from use in layout file as they were causing bild failures due to app.gradle referencing the latest code on master
- Reverted changes made to sample app, activity and xml

samuelo-101 commented:
Hi @MichaelObi my apologies, I have pushed a change in which I removed the use of those 2 new attributes from the sample app. The PR probably failed to build because the gradle dependency currently references the latest code of the main chat-ui repo. All should be well now. Kindly let me know if there are any other issues with the PR. Thanks.

MichaelObi reviewed (approved):
Allowing customisation of the chat message is great. It fixes #16 too.

Thanks

MichaelObi reviewed (commented):
@samuelojo88 I was about to merge this but I have a concern. If you had to remove backgroundSend to make the sample project build, how are we going to set the message row background?

samuelo-101 commented:
@MichaelObi Hi, and thanks for taking the time to review. I understand your concern, however, I tested the 2 new attributes against the sample app locally, with the following changes to the build.gradle file:

Replaced `compile 'co.intentservice.android-chat-ui:chat-ui:0.1.1'` with `compile project(":chat-ui")`

The use of the 2 new attributes in the sample app caused build failures because the gradle dependency referenced the main chat-ui repo (`compile 'co.intentservice.android-chat-ui:chat-ui:0.1.1'`), which does not have these changes merged in yet, and as such, the 2 new attributes could not be used against the sample app. When I changed the gradle dependency to reference the chat-ui module (which contains the changes) it worked. I would, however, recommend an increment of the version minor number before merging this. I hope this answers your question. If there are still some concerns about this, we could also merge this into a separate branch, like a dev-branch, for testing before release.

Thanks.

MichaelObi commented:
Thanks a lot for clarifying @samuelojo88.

Merging this now.

samuelo-101 commented:
@MichaelObi Thank you too. This should allow #16 to be closed too.
